bip-p2qrh.mediawiki

```diff
@@ -50,26 +50,40 @@ The following table is non-exhaustive, but meant to be informative to the averag
+
+Support for FALCON will also be introduced. While its signatures are roughly 4x larger than SQIsign signatures and 20x larger than Schnorr signatures, it will be necessary to have a fallback algorithm should a flaw be found in SQIsign, which is much newer and experimental. FALCON is a more conservative approach to applied lattice cryptography than SQIsign, and its use is approved by NIST. However, even SQIsign signatures are roughly 5x larger than Schnorr signatures. This means, to maintain present transaction throughput, an increase in the witness discount will likely be desired in a QuBit soft fork. That will be specified in a future QuBit BIP.
+
+An increase in the witness discount must not be taken lightly, and must be resistant to applications that might take advantage of this discount such as storage of arbitrary data (commonly known as "inscriptions"), inflating chain state on nodes without a corresponding increase in economic activity. This would not only impact node runners, but those with inscriptions would also have the scarcity of their non-monetary assets affected.
+
+To address this, very specific rules will be applied to spending from the witness stack in SegWit v3 outputs. A fixed signature size will be necessary for spending the output, and the output must be spendable to be considered valid within node consensus. A fixed signature size will also be helpful in disambiguating between signature types without an additional version byte, as SQIsign signatures are substantially smaller than FALCON signatures, and vice versa, and so the correct signature algorithm can be inferred. The public key and signature will be pushed separately to the witness stack. Multiple signatures can be included in order to support multisig applications.
+
+Since only valid signatures can be committed to in a SegWit v3 witness, arbitrary data cannot be added by miners, as that would affect the consensus of their block. Technically if a CRQC were capable of computing a spendable public key that matched arbitrary signature data, this is economically disincentivized due to the cost of that computation, which could prove quite substantial, rather than simply putting the arbitrary data within a Taproot witness, rather than doing the work to meet the requirement for it to be consensus-valid data.
```

Comment:
```suggestion
Since only valid signatures can be committed to in a SegWit v3 witness, arbitrary data cannot be added by miners, as that would affect the consensus of their block. A CRQC operator is economically disincentivized from computing a spendable public key that matched arbitrary signature data due to the cost of that computation. That's because the cost of such a computation could prove quite substantial, rather than simply putting the arbitrary data within a Taproot witness. Doing the work to meet the requirement for it to be consensus-valid data would prove cost-prohibitive.
```

Reply:
Could you break the "that's" into "that is" above?  It will sound more formal.  Otherwise looks good.

Reply:
Done!